src/main/java/org/terasology/adventureassets/damageplayer/DamagePlayerSystem.java

```diff
@@ -45,7 +45,7 @@
     public void onCollide(CollideEvent event, EntityRef entity, DamagePlayerComponent damagePlayerComponent) {
         EntityRef player = event.getOtherEntity();
         if (player.hasComponent(AliveCharacterComponent.class)) {
-            player.send(new CharacterImpulseEvent(JomlUtil.from(event.getNormal()).mul(-1 * damagePlayerComponent.recoil)));
+            player.send(new CharacterImpulseEvent(event.getNormal().mul(-1 * damagePlayerComponent.recoil)));
```

Comment:
That's been like that before already, but yeah...
maybe like this then?
```suggestion
            org.joml.Vector3f dir = new org.joml.Vector3f(event.getNormal());
            player.send(new CharacterImpulseEvent(dir.mul(-1 * damagePlayerComponent.recoil)));
```